### ISSUE TITLE
[clkmgr] Idle signal permanence

### ISSUE THREAD

**Issue by tjaychen** [lowRISC/opentitan#11156]
This is a tracking issue to investigate how to tweak the `idle` signal from each transactional block such that they are more fault resistant. 
The basic idea here is that if a block is "glitched" to idle, it should be able to recover if that's a temporary condition.

This will need to be brought up during the security working group to be further discussed.

**Comment by tjaychen** [comment]
@cdgori 
@moidx 
@felixmiller

**Comment by cdgori** [comment]
This corresponds to the stuff we talked about earlier today right?  Assuming so... I'll elaborate:

We want the idle/ok-to-clock-gate signalling to be as "expansive" as possible.  Something like "state == IDLE" is kind of ideal (but not always possible) - with sparse state we'll get an alert if the state gets glitched, and if the ok-to-gate indicator is glitched, it should at most suppress one clock edge when the indicator gets re-clocked/latched.

(Tim and I talked about having some lag in the response of clkmgr to actually doing the clock gating, since we need at least one clock edge to do this re-latching - and in reality a few cycles before gating probably doesn't hurt anything.)
